lib/auth0/api/v2/users.rb

```diff
@@ -185,12 +185,137 @@ def user_logs(user_id, options = {})
+        end
+
+        # Remove one or more permissions from a specific user.
+        # @see https://auth0.com/docs/api/management/v2#!/Users/delete_permissions
+        #
+        # @param user_id [string] The user_id of the permissions to remove.
+        # @param permissions [array] An array of Permissions to remove.
+        def remove_permissions(user_id, permissions)
+          raise Auth0::MissingUserId, 'Must supply a valid user_id' if user_id.to_s.empty?
+          permissions = validate_permissions_array permissions
+          delete("#{users_path}/#{user_id}/permissions", permissions)
+        end
+
+        # Add one or more permissions from a specific user.
+        # @see https://auth0.com/docs/api/management/v2#!/Users/post_permissions
+        #
+        # @param user_id [string] The user_id of the permissions to add.
+        # @param permissions [array] An array of Permissions to add.
+        def add_permissions(user_id, permissions)
+          raise Auth0::MissingUserId, 'Must supply a valid user_id' if user_id.to_s.empty?
+          permissions = validate_permissions_array permissions
+          post("#{users_path}/#{user_id}/permissions", permissions)
+        end
+
+        # Removes the current Guardian recovery code and generates and returns a new one.
+        # @see https://auth0.com/docs/api/management/v2#!/Users/post_recovery_code_regeneration
+        #
+        # @param user_id [string] The user_id of the recovery codes to regenerate.
+        def generate_recovery_code(user_id)
+          raise Auth0::MissingUserId, 'Must supply a valid user_id' if user_id.to_s.empty?
+          post "#{users_path}/#{user_id}/recovery-code-generation"
+        end
+
+        # Invalidates all remembered browsers for all authentication factors for a specific user.
+        # @see https://auth0.com/docs/api/management/v2#!/Users/post_invalidate_remember_browser
+        #
+        # @param user_id [string] The user_id of the browsers to invalidate.
+        def invalidate_browsers(user_id)
+          raise Auth0::MissingUserId, 'Must supply a valid user_id' if user_id.to_s.empty?
+          post "#{users_path}/#{user_id}/multifactor/actions/invalidate-remember-browser"
+        end
+
         private
 
         # Users API path
         def users_path
           @users_path ||= '/api/v2/users'
         end
+
+        # Check a roles array
+        def validate_roles_array(roles)
+          raise Auth0::InvalidParameter, 'Must supply an array of role names' unless roles.kind_of?(Array)
+          raise Auth0::InvalidParameter, 'Must supply an array of role names' if roles.empty?
+          raise Auth0::InvalidParameter, 'All role names must be strings' unless roles.all? {|role| role.is_a? String}
+        end
+
+        # Check a permissions array
+        def validate_permissions_array(permissions)
+          raise Auth0::InvalidParameter, 'Must supply an array of Permissions' unless permissions.kind_of?(Array)
```

Comment:
Capitalized Permissions here too should be permissions.

Reply:
👆